test/test_bdecode.cpp

```diff
@@ -201,8 +201,7 @@ TORRENT_TEST(dict_null_key)
 	int ret = bdecode(b, b + sizeof(b)-1, e, ec);
 	TEST_EQUAL(ret, 0);
 	TEST_CHECK(e.dict_size() == 1);
-	std::string find_str("a\0b", 3);
-	bdecode_node d = e.dict_find(find_str);
+	bdecode_node d = e.dict_find(string_view("a\0b", 3));
```

Comment:
do you need to say ``string_view`` here? I am pretty sure it has a constructor like this:

```
template <std::size_t N>
string_view(char const (&str)[N]) { ... }
```
Allowing to construct from a string literal, and deducing the length of of it automatically (so you don't have to say 3 either)